tests/test_scatterPlot.py

```diff
@@ -0,0 +1,131 @@
+        flux_meas = flux + rng.normal(scale=np.sqrt(flux*(1 + extendedness)))
+        flux_err = np.sqrt(flux_meas * (1 + extendedness))
+        good = (flux_meas/np.sqrt(flux * (1 + extendedness))) > 3
+        extendedness = extendedness[good]
+        flux = flux[good]
+        flux_meas = flux_meas[good]
+        flux_err = flux_err[good]
+
+        # Configure the plot to show observed vs true mags
+        config = ScatterPlotWithTwoHistsTaskConfig(
+            axisLabels={'x': 'mag', 'y': 'mag meas - ref', 'mag': 'mag'},
+        )
+        config.selectorActions.flagSelector.bands = ['i']
+        config.axisActions.yAction = MagDiff(col1='refcat_flux', col2='refcat_flux')
+        config.nonBandColumnPrefixes.append('refcat')
+        config.sourceSelectorActions.galaxySelector = GalaxyIdentifier
+        config.highSnStatisticSelectorActions.statSelector.threshold = 50
+        config.lowSnStatisticSelectorActions.statSelector.threshold = 20
+        self.task = ScatterPlotWithTwoHistsTask(config=config)
+
+        n = len(flux)
+        self.bands, columns = config.get_requirements()
+        data = {
+            'refcat_flux': flux,
+            'patch': np.zeros(n, dtype=int),
+        }
+
+        # Assign values to columns based on their unchanged default names
+        for column in columns:
+            if column not in data:
+                if column.startswith('detect'):
+                    data[column] = np.ones(n, dtype=bool)
+                elif column.endswith('_flag') or 'Flag' in column:
+                    data[column] = np.zeros(n, dtype=bool)
+                elif column.endswith('Flux'):
+                    config.axisActions.yAction.col1 = column
+                    data[column] = flux_meas
+                elif column.endswith('FluxErr'):
+                    data[column] = flux_err
+                elif column.endswith('_extendedness'):
+                    data[column] = extendedness
+                else:
+                    raise RuntimeError(f'Unexpected column {column} in ScatterPlotWithTwoHistsTaskConfig')
+
+        self.data = pd.DataFrame(data)
+
+    def tearDown(self):
+        if os.path.exists(self.testDir):
+            shutil.rmtree(self.testDir, True)
+        del self.bands
+        del self.data
+        del self.task
+
+    def test_ScatterPlotWithTwoHistsTask(self):
+        result = self.task.run(self.data,
+                               dataId={},
+                               runName='test',
+                               skymap=None,
+                               tableName='test',
+                               bands=self.bands,
+                               plotName='test')
+
+        filename_figure_tmp = os.path.join(self.testDir, "test_scatterPlot.png")
+        result.scatterPlot.savefig(filename_figure_tmp)
+        diff = compare_images(filename_figure_tmp, filename_figure_ref, 0)
+        self.assertIsNone(diff)
```

Comment:
This is likely not going to give a helpful error message.

Reply:
This is the correct way to check/raise: https://github.com/LSSTDESC/skyproj/blob/main/tests/test_plotting.py#L47-L48

I'm sure the tolerance will have to be >0 because there are small differences on different platforms.

Reply:
Well, it passed Jenkins with 0 tolerance.
compare_images returns a human-readable message on failure, and pytest prints this out if the assert fails, so what's the benefit of raising an exception in a test?

Reply:
I triggered a failure, which is not hard to do - having repeatedly done that last week, and this was the error message I got.

```
AssertionError: 'Error: Image files did not match.\n  RMS Value: 11.985321431153329\n  Expected:  \n    /project/kannawad/repo/analysis_drp/tests/test_outputahus36lt/test_scatterPlot.png\n  Actual:    \n    /project/kannawad/repo/analysis_drp/tests/data/test_scatterPlot.png\n  Difference:\n    /project/kannawad/repo/analysis_drp/tests/data/test_scatterPlot-failed-diff.png\n  Tolerance: \n    0' is not None
```

While the message is a bit cryptic, I doubt there can be any more helpful message than having "Image files did not match".

Reply:
Did it print with the `\n`s like that?  Because if you do the matplotlib raise then it'll print the message correctly formatted.

Reply:
It was verbatim, with the `\n`s like that.

<img width="1237" alt="image" src="https://user-images.githubusercontent.com/5103390/159617067-df6d9b55-25e7-4bd9-8a52-ee38776209ea.png">

Reply:
Ok, raise it is then.